tests/foreman/cli/test_container_management.py

```diff
@@ -322,3 +322,75 @@ def test_positive_container_admin_end_to_end_pull(
         # 9. Pull in docker image
         result = container_contenthost.execute(docker_pull_command)
         assert result.status == 0
+
+    def test_negative_pull_content_with_longer_name(
+        self, target_sat, container_contenthost, module_org
+    ):
+        """Verify that long name CV publish when CV & docker repo both have larger name.
+
+        :id: e0ac0be4-f5ff-4a88-bb29-33aa2d874f46
+
+        :steps:
+
+            1. Create Product,docker repo,CV and LCE with long name product / repo
+            2. Sync the repos
+            3. Add repository to CV, Publish, Promote CV to LCE
+            4. Pull in docker image
+
+        :expectedresults:
+
+            1. Long Product, repository, CV and LCE should create successfully
+            2. Sync repository successfully
+            3. Publish & Promote should success
+            4. Can pull in docker images
+
+        :BZ: 2127470
+
+        :customerscenario: true
+        """
+        pattern_postfix = gen_string('alpha', 10).lower()
+
+        product_name = f'containers-{pattern_postfix}'
+        repo_name = f'repo-{pattern_postfix}'
+        lce_name = f'lce-{pattern_postfix}'
+        cv_name = f'cv-{pattern_postfix}'
+
+        # 1. Create Product,docker repo,CV and LCE with long name product / repo
+        logger.info("Creating product with longer name")
+        product = target_sat.cli_factory.make_product_wait(
+            {'name': product_name, 'organization-id': module_org.id}
+        )
+        logger.info("Creating docker repo with longer name")
+        repo = _repo(product['id'], name=repo_name, upstream_name=CONTAINER_UPSTREAM_NAME)
+        # 2. Sync the repos
+        target_sat.cli.Repository.synchronize({'id': repo['id']})
+
+        logger.info("Creating lce, cv, add repos to cv, publish and promote")
+        lce = target_sat.cli_factory.make_lifecycle_environment(
+            {'name': lce_name, 'organization-id': module_org.id}
+        )
+        cv = target_sat.cli_factory.make_content_view(
+            {'name': cv_name, 'composite': False, 'organization-id': module_org.id}
+        )
+        ContentView.add_repository({'id': cv['id'], 'repository-id': repo['id']})
+        # 3. Publish & Promote should success
+        ContentView.publish({'id': cv['id']})
+        cv = ContentView.info({'id': cv['id']})
+        ContentView.version_promote(
+            {'id': cv['versions'][0]['id'], 'to-lifecycle-environment-id': lce['id']}
+        )
+        # 4. Can pull in docker images
+        podman_pull_command = (
+            f"podman pull --tls-verify=false {target_sat.hostname}/{module_org.label.lower()}-"
+            f"{product.label.lower()}-{repo_name}"
+        )
```

Comment:
I believe we should test the repo published in the long-named CV and promoted to the long-named LCE at this point (not the basic repo in Library).
```suggestion
        podman_pull_command = (
            f"podman pull --tls-verify=false {target_sat.hostname}/{module_org.label.lower()}-"
            f"{lce_name}-{cv_name}-{product_name}-{repo_name}"
        )
```